tests/test_smart_completion_public_schema_only.py

```diff
@@ -553,3 +553,73 @@ def test_columns_before_keywords(completer, complete_event):
     assert completions.index(column) < completions.index(keyword)
 
 
+def test_wildcard_column_expansion(completer, complete_event):
+    sql = 'SELECT * FROM users'
+    pos = len('SELECT *')
+
+    completions = completer.get_completions(
+        Document(text=sql, cursor_position=pos), complete_event)
+
+    col_list = 'id, email, first_name, last_name'
+    expected = [Completion(text=col_list, start_position=-1,
+                          display='*', display_meta='columns')]
+
+    assert expected == completions
+
+
+def test_wildcard_column_expansion_with_alias_qualifier(completer, complete_event):
+    sql = 'SELECT u.* FROM users u'
+    pos = len('SELECT u.*')
+
+    completions = completer.get_completions(
+        Document(text=sql, cursor_position=pos), complete_event)
+
+    col_list = 'id, u.email, u.first_name, u.last_name'
+    expected = [Completion(text=col_list, start_position=-1,
+                          display='*', display_meta='columns')]
+
+    assert expected == completions
+
+
+def test_wildcard_column_expansion_with_table_qualifier(completer, complete_event):
+    sql = 'SELECT users.* FROM users'
+    pos = len('SELECT users.*')
+
+    completions = completer.get_completions(
+        Document(text=sql, cursor_position=pos), complete_event)
+
+    col_list = 'id, users.email, users.first_name, users.last_name'
+    expected = [Completion(text=col_list, start_position=-1,
+                          display='*', display_meta='columns')]
+
+    assert expected == completions
+
+def test_wildcard_column_expansion_with_two_tables(completer, complete_event):
+    sql = 'SELECT * FROM "select" JOIN users u ON true'
+    pos = len('SELECT *')
+
+    completions = completer.get_completions(
+        Document(text=sql, cursor_position=pos), complete_event)
+
+    # The order of the tables is indeterministic, so allwo both possibilities
```

Comment:
So I'm pretty sure the reason the table order is indeterministic here is that you now have `populate_scoped_cols` returning a dict, and `get_column_matches` is now iterating over the items of that dict, and of course the order of keys in a dict is non-deterministic. It'd be nice to fix this issue there, instead of having the more flexible test here. At least for the user it'd be nice to have columns suggested in a predictable manner. I think you could have  `populate_scoped_cols` return a ordered dict instead, if such a thing is possible with defaultdict.